web-app/src/app/tabs/race/race-graph/race-graph.component.ts

```diff
@@ -0,0 +1,217 @@
+import { Component, OnInit } from '@angular/core';
+import { MatRadioModule } from '@angular/material/radio';
+
+import { LapService } from '../../../_services/lap.service';
+import { LapData } from '../../../_objects/lapData';
+
+import * as CanvasJS from '../../../../assets/thirdParty/canvasjs.min.js';
+
+@Component({
+  selector: 'app-race-graph',
+  templateUrl: './race-graph.component.html',
+  styleUrls: ['./race-graph.component.css']
+})
+export class RaceGraphComponent implements OnInit {
+
+  lapTimeArray: GraphDataPoint[] = [];
+  totalPowerInArray: GraphDataPoint[] = [];
+  totalPowerOutArray: GraphDataPoint[] = [];
+  netPowerOutArray: GraphDataPoint[] = [];
+  distanceArray: GraphDataPoint[] = [];
+  amphoursArray: GraphDataPoint[] = [];
+  averagePackCurrentArray: GraphDataPoint[] = [];
+  batterySecondsRemainingArray: GraphDataPoint[] = [];
+  lapData: LapData[] = [];
+
+  graphOptions: string[] = ['Time', 'Power', 'Distance', 'Amp Hours', 'Current', 'Battery' ];
+  currLap: number;
+  graphHeader: string;
+  selectedGraph = 'Distance';
+  chart: CanvasJS.Chart;
+
+  constructor(private lapService: LapService) {
+  }
+
+   ngOnInit() {
+    this.currLap = 0;
+    this.initializeDataArrays();
+
+    this.lapService.lapData$.subscribe(
+      (data: LapData[]) => {
+        this.addLapData(data[0]) // most recent data point is at array beginning
+        this.updateChart();
+      });
+
+    this.chart = new CanvasJS.Chart('raceGraph', {
+      exportEnabled: true,
+      theme: 'dark1',
+      backgroundColor: 'transparent',
+      title: {
+        text : '',
+        fontSize: 20
+      },
+      data: [{
```

Comment:
Can we just initialize data to be an empty array? And then get rid of all the individual points.

```
data: [],
axisX:....
```